internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/service/VeracodeScanner.java

```diff
@@ -0,0 +1,304 @@
+                    "the scan artifacts are empty in the request. ";
+            responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                    message), HttpStatus.BAD_REQUEST);
+            CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.ERROR, null,
+                    null);
+            CallbackUtil.persistScanLog(scanContext.getJobId(), message, LogType.ERROR);
+        }
+        return responseEntity;
+    }
+
+    /**
+     * Stop the last scan for a given application.
+     *
+     * @param scanRequest Object that represent the required information for tha scanner operation
+     * @return ResponseEntity with status of the request
+     */
+    @Override
+    public ResponseEntity cancelScan(ScannerScanRequest scanRequest) {
+
+        ResponseEntity responseEntity;
+        String result;
+
+        try {
+            result = uploadAPIWrapper.deleteBuild(scanContext.getAppId());
+            CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.CANCELED, null,
+                    null);
+            if (VeracodeResultProcessor.isOperationProceedWithoutError(result)) {
+                responseEntity = new ResponseEntity<>(HttpStatus.ACCEPTED);
+            } else {
+                String message = "Error occured while deleting the last scan of the application : "
+                        + scanRequest.getAppId();
+                responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                        message), HttpStatus.BAD_REQUEST);
+                CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.ERROR, null,
+                        null);
+                CallbackUtil.persistScanLog(scanContext.getJobId(), message, LogType.ERROR);
+            }
+        } catch (IOException e) {
+            String message = "Error occured while deleting the last scan of the application : "
+                    + scanRequest.getAppId() + VeracodeResultProcessor.getFullErrorMessage(e);
+            responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.INTERNAL_SERVER_ERROR.value(),
+                    message), HttpStatus.INTERNAL_SERVER_ERROR);
+            CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.ERROR, null,
+                    null);
+            CallbackUtil.persistScanLog(scanContext.getJobId(), message, LogType.ERROR);
+        }
+        return responseEntity;
+    }
+
+    /**
+     * Build the upload wrap for Upload API.
+     *
+     * @param options represents the Veracode credentials and other required configurations
+     * @return Veracode Upload API Wrapper
+     * @throws UnsupportedEncodingException
+     */
+    private UploadAPIWrapper getUploadAPIWrapper(VeracodeCommand.Options options) throws UnsupportedEncodingException {
+        uploadAPIWrapper = new UploadAPIWrapper();
```

Comment:
Multiple calls to this method will mask the previous instance. IMO this is not a good practice. 

IMO `getUploadAPIWrapper` should be in a separate Util (`VeracodeAPIUtil`). We should not have class level variables holding `UploadAPIWrapper` references anywhere in the code.  Instead, before every upload operations we should get a new instance of `UploadAPIWrapper`. WDYT?

`VeracodeAPIUtil` can be initialized with the `Credentials` during the configuration loading phase / initialization process of the scanner. 

Same comment applies to other `get*APIWrapper` methods.